Rek 268 ecklc deploy (#240)
OHS-Hosting-Infrastructure/infrastructure: docs/how-we-work/deploy-eclkc.md

1. Sam then runs `drush cr` on the stage1 machine to clear the Drupal cache.
1. Once that is complete, Sam runs the deployment job ([GitHub Stage Branch Switch](https://jenkins.eclkc.info/job/Github%20Stage%20Branch%20Switch/configure)), which runs on both Stage1 and Stage2. This job:

    * Pulls the code from the repo to both staging servers
    * Runs composer install (to install any new modules/dependencies)
    * Refreshes cache
    * Imports new config from repo to update db
    * If the modules contain updates or scripts that may affect the db, those scripts are run to update db
    * Flushes Drupal cache again
    * Restarts Varnish

    Sam specifies which branch it runs on (usually specifies the develop branch). This is a feature they want to keep.

1. Sam takes a look at the Jenkins console for the job to verify that there are no warnings.
1. Everything is now deployed into Stage with a production copy of the database.
1. Sam does a sweep of the staging site [stage.eclkc.info](https://stage.eclkc.info). He makes sure there are no major, obvious issue. When he's done, he alerts Bhuvan and one of them set the status of the issue to "Testing".

> 1. Bhuvan tests each of the issues in the the Fix Version queue in the "In Review" state. He will also complete regression testing.

Reviewer: Now the issues are in "Testing" because we move them there on the previous step.

1. The team will do User Acceptance Testing for the issues that require it. This is relatively infrequent, and are typically UI-related. Usually this UAT looks like a demo or training rather than formal UAT. If there is a large functional change, they may have stakeholders validate against a list of requirements.
1. If there are additional updates, they can run them at any time in staging.

> Depending on whether they want to install or remove a module, they have to complete various certain steps in a certain order. If they are removing, they first remove the code then the module. If they are installing, they install the module first and then add the code.

Reviewer: The other way around:
```suggestion
Depending on whether they want to install or remove a module, they have to complete various certain steps in a certain order. If they are removing, they first disable the module, uninstall and then remove the code. If they are installing, they pull the code and then install the module with its config.
```

> ## Production

Reviewer: These steps don't include backups, Lifeboat sync and traffic switch that is performed during deployments.

Author: Good point, I'll add that in from here: https://github.com/OHS-Hosting-Infrastructure/infrastructure/blob/main/docs/runbooks/how-to-support-eclkc-drupal-deploy.md

Everything in production is manual. Sam does all the work himself, but hopes to automate it.

> Previously, you would make all parameter, module, and configuration changes for each environment via the Drupal UI. Now, Drupal has **configuration management**, which handles the parameters, layout, and other configs across environments. They use this UI.

Reviewer: ```suggestion
Previously, in Drupal 7, you would make all parameter, module, and configuration changes for each environment via the Drupal UI. Now, Drupal has **configuration management**, which handles the parameters, layout, and other configs across environments. They use this UI. The configuration can be exported, stored in the repository and imported in a different environment.
```

> To uninstall a module, Sam disables it in the UI or through the command line and then runs composer and the code for the module is removed.

Reviewer: ```suggestion
To uninstall a module, Sam disables it in the UI or through the command line (via drush command or config import) and then runs composer and the code for the module is removed.
```
